pkg/k8s/api.go

```diff
@@ -245,6 +245,27 @@ func GetPodStatus(pod corev1.Pod) string {
 	return reason
 }
 
+// CheckForEnv checks for the given key in the given Container env variables
+func CheckForEnv(container *corev1.Container, envKey string) bool {
+	var presence bool
+	for _, env := range container.Env {
+		if env.Name == envKey {
+			presence = true
```

Comment:
This is actually going to change, once #5608 is merged. The plan is to check only for annotation that the injector has added instead of explicitly checking for the environment variable. WDYT?